crypto/CMakeLists.txt

```diff
@@ -110,12 +110,19 @@ else()
   elseif(APPLE)
     set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/mac-${ARCH}/crypto/)
   elseif(UNIX)
-    set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/linux-${ARCH}/crypto/)
+    if(${ARCH} STREQUAL "generic")
+      message(STATUS "Detected generic linux platform. No assembly files will be included.")
+    else()
+      set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/linux-${ARCH}/crypto/)
```

Comment:
There are two patterns we can make use of here:
1. Assume that if `ARCH` is not `generic` we have pre-generated source for it. Or
2. Explicitly check for `ARCH`s that we have pre-generated source for.

Or is this filtered somewhere else?

Reply:
I could change this to instead see if a directory exists for the `ARCH` defined at `${GENERATE_CODE_ROOT}/linux-${ARCH}/crypto` and only set `ASSEMBLY_SOURCE` if the generated files exist for it. 

The current behavior assumes that generated source exists for any `ARCH` other than `generic` based on the checks we do [here](https://github.com/awslabs/aws-lc/blob/88d124deee4e948a78d8e710d35af5fc18259a82/CMakeLists.txt#L588-L611). Any `CMAKE_SYSTEM_PROCESSOR` other than `generic` or other entries that we generate build files for, will result in a fatal error.

Reply:
Thanks. We explicitly check that an architecture is in a pre-defined list (modeled through the if-else). So, should not end up here with a `ARCH` that is unknown to us.